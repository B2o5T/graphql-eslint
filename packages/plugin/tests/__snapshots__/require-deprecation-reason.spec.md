// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Invalid #1 1`] = `
#### ⌨️ Code

       1 |         type A {
       2 |           deprecatedWithoutReason: String @deprecated
       3 |           deprecatedWithReason: String @deprecated(reason: "Reason")
       4 |           notDeprecated: String
       5 |         }
       6 |
       7 |         enum TestEnum {
       8 |           item1 @deprecated
       9 |           item2 @deprecated(reason: "Reason")
      10 |         }
      11 |
      12 |         interface TestInterface {
      13 |           item1: String @deprecated
      14 |           item2: Number @deprecated(reason: "Reason")
      15 |           item3: String
      16 |           item4: String @deprecated(reason: "")
      17 |           item5: String @deprecated(reason: "  ")
      18 |         }

#### ❌ Error 1/5

      1 |         type A {
    > 2 |           deprecatedWithoutReason: String @deprecated
        |                                            ^^^^^^^^^^ Deprecation reason is required for field "deprecatedWithoutReason" in type "A".
      3 |           deprecatedWithReason: String @deprecated(reason: "Reason")

#### ❌ Error 2/5

      7 |         enum TestEnum {
    > 8 |           item1 @deprecated
        |                  ^^^^^^^^^^ Deprecation reason is required for field "item1" in type "TestEnum".
      9 |           item2 @deprecated(reason: "Reason")

#### ❌ Error 3/5

      12 |         interface TestInterface {
    > 13 |           item1: String @deprecated
         |                          ^^^^^^^^^^ Deprecation reason is required for field "item1" in type "TestInterface".
      14 |           item2: Number @deprecated(reason: "Reason")

#### ❌ Error 4/5

      15 |           item3: String
    > 16 |           item4: String @deprecated(reason: "")
         |                          ^^^^^^^^^^ Deprecation reason is required for field "item4" in type "TestInterface".
      17 |           item5: String @deprecated(reason: "  ")

#### ❌ Error 5/5

      16 |           item4: String @deprecated(reason: "")
    > 17 |           item5: String @deprecated(reason: "  ")
         |                          ^^^^^^^^^^ Deprecation reason is required for field "item5" in type "TestInterface".
      18 |         }
`;
