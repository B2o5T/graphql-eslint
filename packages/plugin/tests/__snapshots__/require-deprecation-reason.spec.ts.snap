// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Error 1/5

   1 |
   2 |         type A {
>  3 |           deprecatedWithoutReason: String @deprecated
     |                                            ^^^^^^^^^^ Directive "@deprecated" must have a reason!
   4 |           deprecatedWithReason: String @deprecated(reason: "Reason")
   5 |           notDeprecated: String
   6 |         }
   7 |
   8 |         enum testEnum {
   9 |           item1 @deprecated
  10 |           item2 @deprecated(reason: "Reason")
  11 |         }
  12 |
  13 |         interface testInterface {
  14 |           item1: String @deprecated
  15 |           item2: Number @deprecated(reason: "Reason")
  16 |           item3: String
  17 |           item4: String @deprecated(reason: "")
  18 |           item5: String @deprecated(reason: "  ")
  19 |         }
  20 |       

Error 2/5

   1 |
   2 |         type A {
   3 |           deprecatedWithoutReason: String @deprecated
   4 |           deprecatedWithReason: String @deprecated(reason: "Reason")
   5 |           notDeprecated: String
   6 |         }
   7 |
   8 |         enum testEnum {
>  9 |           item1 @deprecated
     |                  ^^^^^^^^^^ Directive "@deprecated" must have a reason!
  10 |           item2 @deprecated(reason: "Reason")
  11 |         }
  12 |
  13 |         interface testInterface {
  14 |           item1: String @deprecated
  15 |           item2: Number @deprecated(reason: "Reason")
  16 |           item3: String
  17 |           item4: String @deprecated(reason: "")
  18 |           item5: String @deprecated(reason: "  ")
  19 |         }
  20 |       

Error 3/5

   1 |
   2 |         type A {
   3 |           deprecatedWithoutReason: String @deprecated
   4 |           deprecatedWithReason: String @deprecated(reason: "Reason")
   5 |           notDeprecated: String
   6 |         }
   7 |
   8 |         enum testEnum {
   9 |           item1 @deprecated
  10 |           item2 @deprecated(reason: "Reason")
  11 |         }
  12 |
  13 |         interface testInterface {
> 14 |           item1: String @deprecated
     |                          ^^^^^^^^^^ Directive "@deprecated" must have a reason!
  15 |           item2: Number @deprecated(reason: "Reason")
  16 |           item3: String
  17 |           item4: String @deprecated(reason: "")
  18 |           item5: String @deprecated(reason: "  ")
  19 |         }
  20 |       

Error 4/5

   1 |
   2 |         type A {
   3 |           deprecatedWithoutReason: String @deprecated
   4 |           deprecatedWithReason: String @deprecated(reason: "Reason")
   5 |           notDeprecated: String
   6 |         }
   7 |
   8 |         enum testEnum {
   9 |           item1 @deprecated
  10 |           item2 @deprecated(reason: "Reason")
  11 |         }
  12 |
  13 |         interface testInterface {
  14 |           item1: String @deprecated
  15 |           item2: Number @deprecated(reason: "Reason")
  16 |           item3: String
> 17 |           item4: String @deprecated(reason: "")
     |                          ^^^^^^^^^^ Directive "@deprecated" must have a reason!
  18 |           item5: String @deprecated(reason: "  ")
  19 |         }
  20 |       

Error 5/5

   1 |
   2 |         type A {
   3 |           deprecatedWithoutReason: String @deprecated
   4 |           deprecatedWithReason: String @deprecated(reason: "Reason")
   5 |           notDeprecated: String
   6 |         }
   7 |
   8 |         enum testEnum {
   9 |           item1 @deprecated
  10 |           item2 @deprecated(reason: "Reason")
  11 |         }
  12 |
  13 |         interface testInterface {
  14 |           item1: String @deprecated
  15 |           item2: Number @deprecated(reason: "Reason")
  16 |           item3: String
  17 |           item4: String @deprecated(reason: "")
> 18 |           item5: String @deprecated(reason: "  ")
     |                          ^^^^^^^^^^ Directive "@deprecated" must have a reason!
  19 |         }
  20 |       
`;
